pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/Exceptions.java

```diff
@@ -48,4 +48,15 @@ public static String toString(Exception ex) {
         return sw.toString();
     }
 
+    public static boolean isExceptionPresentInChain(Throwable error, Class type) {
```

Comment:
We could allow for multiple types here:

```suggestion
    public static boolean areExceptionsPresentInChain(Throwable error, Class types...) {
```